### PR TITLE
Fixes for use with filewriter

### PR DESCRIPTION
**Pull request by mattclarke**
These are the fixes for date that allows it to work with the filewriter.
We need to check this doesn't break daquiri before merging.

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -3,86 +3,102 @@
...
 
-        os.rename(os.path.join(self.folder_name, "CMakeLists.txt"),
-                  os.path.join(self.folder_name, "CMakeLists_original.txt"))
-        shutil.copy("CMakeLists.txt", os.path.join(self.folder_name, "CMakeLists.txt"))
-        shutil.copy("date-config.cmake", os.path.join(self.folder_name, "date-config.cmake"))
+    def source(self):
+        self.run("git clone https://github.com/HowardHinnant/date.git")
+        with tools.chdir("./{}".format(self.name)):
+            self.run("git checkout 54e8516af223670b75d7a17c2538c6e6d0843c1f .")
+            self._source_subfolder = os.getcwd()
```
Comment (amues) [inline conanfile.py]: This line is not executed if the source code has already available (e.g. if the package has been built before with different settings and options). This is causing problems in the Jenkins build, as if `_source_subfolder` is not defined correctly, the _include_ directory is not copied to the created package. My suggestion is to use something like `os.path.join(self.source_folder, "date")` where that variable is needed, instead.

File: conanfile.py [hunk 2]
```diff
@@ -3,86 +3,100 @@
...
-
-    def configure(self):
-        self.requires.add("libcurl/7.56.1@bincrafters/stable", private=False)
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
         if self.settings.os == "Macos":
             self.options["libcurl"].darwin_ssl = False
 
+    def configure(self):
```
Comment (martukas) [inline conanfile.py]: Any plan to get this working as well?

Comment (amues) [reply conanfile.py]: How does it fail on your machine?

**Comment by martukas** [comment]
Ok, this works for daquiri, so I'm gonna say it's good to merge.

**Pull request merged**